src/test/java/de/tum/in/www1/artemis/ProgrammingExerciseTestCaseServiceTest.java

```diff
@@ -160,4 +162,82 @@ public void shouldRecalculateScoreBasedOnTestCasesWeight() {
         assertThat(result.getScore()).isEqualTo(expectedScore);
     }
 
+    @Test
+    public void shouldRemoveTestsWithAfterDueDateFlagIfDueDateHasNotPassed() {
+        // Set programming exercise due date in future.
+        programmingExercise.setDueDate(ZonedDateTime.parse("2999-06-30T12:30:40Z[UTC]"));
```

Comment:
Opinion: I think using ZonedDateTime.now().plus.... is more readable and still gives you a date in the future.

Reply:
Thanks! I didn't know that.